File: Stack.cpp
#include<iostream>
//#include<bits/stdc++.h>
#define size 5
using namespace std;

int top = -1, ele;
int stack[size];

int isFull(){
    if(top == size-1){
        return 1;
    }
    else{
        return 0;
    }
}

int isEmpty(){
    if(top==-1){
        return 1;
    }
    else{
        return 0;
    }
}

void push(int element){
    if(isFull()){
        cout<<"the stack is full sorry!"<<endl;
    }
    else{
        top++;
        stack[top] = element;
        cout<<"one element is pushed succefully"<<endl;;
    }
}

int pop(){
    int element;
    if(isEmpty()){
        cout<<"the stack is empty, Sorry!"<<endl;
    }
    else{
        return stack[top--];
        cout<<"one Elemtn poped succesfully"<<endl;
    }
}

int select(){
    if(isEmpty()){
        return 0;
    }
    else{
        return top;
    }
}

void traverse(){
    int i;
    if(isEmpty()){
        cout<<"the stack is Empty"<<endl;
    }
    else{
        for(i=0; i<=top; i++){
            cout<<stack[i]<<endl;
        }
    }
}

int main(){
    int t, temp;
    t = 1;
    while (t--){
        int test;
        cout<<"-----choice is your------"<<endl;
        cout<<"Enter 1 for: push; 2 for: pop; 3 for: traverse; 4 for: select? "<<endl;
        cin>>test;
        switch(test){
            case 1:
                cout<<"Enter element for push data: "<<endl;
                cin>> ele;
                push(ele);
                break;
            case 2:
                ele = pop();
                if(ele == 0){
                    cout<<"Stack is underflow!"<<endl;
                }
                else{
                    cout<<"The deleted item is: "<<ele<<endl;
                }
                break;
            case 3:
                traverse();
                break;
            case 4:
                select();
                break;
            default:
                cout<<"invlid Entery";
                break;
        }
        cout<<"are you want to close the programe??"<<endl;
        cout<<"Enter 1 to run agin and 0 to terminate: ";
        cin>>temp;
        t = temp;
    }
    
}

File: Stack.java
import java.util.*;

import javax.lang.model.element.Element;
public class Stack{
    static int stack_arr[], top = -1, size, item;
    static Scanner s = new Scanner(System.in);

    static{
        Stack.create();
    }

    public static void main(String[] args) {
        boolean t=true, temp;
        int choiche;

        while(t){
            System.out.println("1. push");
            System.out.println("2. pop");
            System.out.println("3. traverse");
            System.out.println("4. pick");
            //System.out.println("5. exit");
            System.out.println("Enter your choiche: ");
            choiche = Stack.s.nextInt();

            switch(choiche){
                case 1:
                    System.out.println("Enter element for push: ");
                    item = Stack.s.nextInt();
                    Stack.push(item);
                break;

                case 2:
                    item = pop();
                    if(item == 0){
                        System.out.println("the Stack is underfllow ");
                    }
                    else{
                        System.out.println("poped succefull "+item);
                    }
                break;

                case 3:
                traverse();
                break;
                case 4:
                    item = pick();
                    if(item == 0){
                        System.out.println("The stack is underflow");
                    }
                    else{
                        System.out.println("peek is :"+item);
                    }

                break;
                case 5:
                System.exit(1);
                default:
                System.out.println("invalid key pressed! sorry");
                break;
            }
            
            System.out.print("1. for run agin & 0 for quit: ");
            choiche = Stack.s.nextInt();

            switch(choiche){
                case 1:
                    temp = true;
                    t = temp;
                    break;
                case 0:
                    temp = false;
                    t = temp;
                    break;
                default:
                    System.out.println("invalid Enter !\b");
                    break;
            }
        }
    }
    static void create(){
        System.out.print("Enter size of Stack: ");
        size = Stack.s.nextInt();
        Stack.stack_arr = new int[size];
        System.out.println("The stack is create succesfully : "+size);
    }

    static void push(int item){
        if(Stack.isFull()){
            System.out.println("The stack is full");
        }
        else{
            Stack.stack_arr[++top] = Stack.item;
        }
    }

    static boolean isFull(){
        if(top == size-1){
            return true;
        }
        else{
            return false;
        }
    }

    static int pop(){
        if(Stack.isEmpty()){
            return 0;
        }
        else{
            return stack_arr[top--];
        }
    }

    static boolean isEmpty(){
        if(top == -1){
            return true;
        }
        else{
            return false;
        }
    }

    static void traverse(){
        if(Stack.isEmpty()){
            System.out.println("the sack is empty");
        }
        else{

            System.out.println("...start traversing data from stack.....");
            for(int i = top; i>=0; i--){
                System.out.println("stack data: " +stack_arr[i]);
            }
        }
    }

    static int pick(){
        if(Stack.isEmpty()){
            return 0;
        }
        else{
            return stack_arr[top];
        }
    }
}